### PR TITLE
fix: adds support for printing number and boolean messageKey value type

### LINKED ISSUE #433: Pino Pretty does not output numbers but only strings.
I have a very simple log file with as below `app.log`
```
{"level":30,"time":"2023-06-19T04:01:56.478Z","pid":2484,"hostname":"YMLTITD2227","msg":77}
{"level":30,"time":"2023-06-19T04:01:56.478Z","pid":2484,"hostname":"YMLTITD2227","msg":"42"}
```

Notice that first message is number 77, while the other is string 42. If I do `cat app.log | npx pino-pretty`, I get the below output
```
[12:01:56.478] INFO (2484):
[12:01:56.478] INFO (2484): 42
```

It does NOT display the numbers, but only the strings. I was hoping that pino-pretty should be able to interpret numbers as strings.

### PR DESCRIPTION
**Pull request by rjmohammad**
closes #433

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: test/basic.test.js [hunk 1]
```diff
@@ -118,6 +118,51 @@ test('basic prettifier tests', (t) => {
...
+        t.equal(
+          formatted,
+          `[${formattedEpoch}] INFO (${pid}): baz\n`
+        )
+        cb()
+      }
+    }))
+    log.info('baz')
+  })
+  t.test('can print message key value when its a number', (t) => {
```
Comment (jsumners) [inline test/basic.test.js:136]:
```suggestion
  })
  
  t.test('can print message key value when its a number', (t) => {
```

File: test/basic.test.js [hunk 2]
```diff
@@ -118,6 +118,51 @@ test('basic prettifier tests', (t) => {
...
+        t.equal(
+          formatted,
+          `[${formattedEpoch}] INFO (${pid}): 42\n`
+        )
+        cb()
+      }
+    }))
+    log.info(42)
+  })
+  t.test('can print message key value when its a string', (t) => {
```
Comment (jsumners) [inline test/basic.test.js:151]:
```suggestion
  })
  
  t.test('can print message key value when its a boolean', (t) => {
```

File: test/basic.test.js [hunk 3]
```diff
@@ -118,6 +118,51 @@ test('basic prettifier tests', (t) => {
...
+        t.equal(
+          formatted,
+          `[${formattedEpoch}] INFO (${pid}): true\n`
+        )
+        cb()
+      }
+    }))
+    log.info(true)
+  })
   t.test('can use different message keys', (t) => {
```
Comment (jsumners) [inline test/basic.test.js:166]:
```suggestion
  })
  
  t.test('can use different message keys', (t) => {
```

File: test/basic.test.js [hunk 4]
```diff
@@ -118,6 +118,51 @@ test('basic prettifier tests', (t) => {
...
+      write (chunk, enc, cb) {
+        const formatted = pretty(chunk.toString())
+        t.equal(
+          formatted,
+          `[${formattedEpoch}] INFO (${pid}): true\n`
+        )
+        cb()
+      }
+    }))
+    log.info(true)
```
Comment (jsumners) [inline test/basic.test.js:164]: I wonder what `log.info(0)` or `log.info(1)` would produce here.

**Review by jsumners — CHANGES_REQUESTED** [review]

File: test/basic.test.js (hunk 4 above)
Comment (rjmohammad) [reply test/basic.test.js:182]: I've added a possible fix for 0.

What is the expectation for `log.info(null)` or `log.info(undefined`)?

Comment (jsumners) [reply test/basic.test.js:182]: Only `null` is valid json. So I imagine we'd just print the word "null". But that can be a separate PR.

**Review by jsumners — APPROVED** [review]

**Pull request merged**